docs/next/guides/building-and-testing/plugins.md

```diff
@@ -1,242 +1,175 @@
 
-There are two types of plugins: internal and external. While both extend Handsontable's functionality, the former is incorporated into the Handsontable build and the latter needs to be included from a separate file.
+### 1. Prerequisites
 
-Regardless of which plugin you are going to build, using a template will save you lots of time.
+You need to import the following things:
+- `BasePlugin` - contains the basic interface to work within Handsontable's lifecycle,
+- `registerPlugin` - utility to register the plugin in Handsontable plugins registry.
 
-#### Internal plugin
+Both provide the basic interface necessary to create a reusable plugin.
+
+```js
+import { BasePlugin, registerPlugin } from 'handsontable/plugins';
+```
+
+### 2. Extend BasePlugin
+Extending our `BasePlugin` is a recommended way to start creating your next extension.
+In this built-in interface, we care about many factors such as backward compatibility, memory leaks preventions and proper binding plugin's instance with Handsontable.
 
-The source code of this template is [available on GitHub](https://github.com/handsontable/handsontable-skeleton/tree/master/plugins/internal). Start off by cloning the entire project or copying the code to your application.
 
 ```js
-// You need to import the BasePlugin class in order to inherit from it.
-import BasePlugin from './../../_base';
-import {registerPlugin} from './../../../plugins';
-
-/**
- * @plugin InternalPluginSkeleton
- * Note: keep in mind, that Handsontable instance creates one instance of the plugin class.
- *
- * @description
- * Blank plugin template. It needs to inherit from the BasePlugin class.
- */
-class InternalPluginSkeleton extends BasePlugin {
-
-  // The argument passed to the constructor is the currently processed Handsontable instance object.
+export class CustomPlugin extends BasePlugin {
+  /**
+   * Define an unique plugin's key.
+   * Handsontable will use that string to register the plugin under that alias.
+   * Moreover, you might also use the same parameter to recognize the plugin's
+   * options passed through the setting object at Handsontable initialization.
+   *
+   * @returns {string}
+   */
+  static get PLUGIN_KEY() {
+    return 'customPlugin';
+  }
+
+  /**
+   * Extend the default constructor and define internal properties for your plugin.
+   *
+   * @param {Handsontable} hotInstance
+   */
   constructor(hotInstance) {
     super(hotInstance);
 
     // Initialize all your public properties in the class' constructor.
-    /**
-     * yourProperty description.
-     *
-     * @type {String}
-     */
-    this.yourProperty = '';
-    /**
-     * anotherProperty description.
-     * @type {Array}
-     */
-    this.anotherProperty = [];
+    this.configuration = null;
   }
 
   /**
    * Checks if the plugin is enabled in the settings.
    */
   isEnabled() {
-    return !!this.hot.getSettings().internalPluginSkeleton;
+    return !!this.hot.getSettings()[CustomPlugin.PLUGIN_KEY];
```

Comment:
For the tutorial we should make it obvious that it is enabled if `true` or has settings. May add the explanation to the comment above
```suggestion
    const pluginSettings = this.hot.getSettings()[CustomPlugin.PLUGIN_KEY];
    return pluginSettings === true || Object.isObject(pluginSettings);
```

Reply:
I got the point, but I see the two problems: 
1) `Object.isObject` ❤️😄 great idea but not exists in ES
2) We do not export modularized `handsontable/helpers` where we have our own helper of `isObject` 🙄

Reply:
1. Hold my 🍺 we have to fix W3C spec first 😄 
2. Maybe lodash `_.isObject` then?
3. We may also add plugin method `validateConfig` and place some more detailed asserts there

Reply:
Yes, I thought about the third option too. Now there will be the "getUnifiedConfig" method.